test/e2e/tls_util_test.go

```diff
@@ -142,3 +159,73 @@ func TestOnlyAppSecretExist(t *testing.T) {
 		t.Fatalf("expect %v, but got %v", tlsutil.ErrCANotFound.Error(), err.Error())
 	}
 }
+
+// TestOnlyCAExist ensures that at the case where only the CA exists in the cluster;
+// GenerateCert can retrieve the CA and uses it to create a new application secret.
+func TestOnlyCAExist(t *testing.T) {
+	f := framework.Global
+	ctx := f.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, err = f.KubeClient.CoreV1().ConfigMaps(namespace).Create(caConfigMap)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = f.KubeClient.CoreV1().Secrets(namespace).Create(caSecret)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cg := tlsutil.NewSDKCertGenerator(f.KubeClient)
+	// Use Pod as a dummy runtime object for the CR input of GenerateCert().
+	mCR := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind: crKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crName,
+			Namespace: namespace,
+		},
+	}
```

Comment:
You might want to make a helper for this, now or in a later PR, since we're using the dummy runtime object in all test cases.

Reply:
yeah, I will do another round of refactoring after finishing all the test cases.